platform/qt/src/local_glyph_rasterizer.cpp

```diff
@@ -0,0 +1,80 @@
+#include <mbgl/text/local_glyph_rasterizer.hpp>
+#include <mbgl/util/i18n.hpp>
+
+#include <QtCore/QFile>
+#include <QtGui/QFont>
+#include <QtGui/QImage>
+#include <QtGui/QPainter>
+#include <qglobal.h>
+
+namespace mbgl {
+
+class LocalGlyphRasterizer::Impl {
+public:
+    Impl(const optional<std::string> fontFamily_);
+
+    bool isConfigured() const;
+
+    optional<std::string> fontFamily;
+    QFont font;
+};
+
+LocalGlyphRasterizer::Impl::Impl(const optional<std::string> fontFamily_)
+    : fontFamily(fontFamily_) {
+    if (isConfigured()) {
+        font.setFamily(QString::fromStdString(*fontFamily));
+        font.setPixelSize(24);
+    }
+}
+
+bool LocalGlyphRasterizer::Impl::isConfigured() const {
+    return fontFamily.operator bool();
+}
+
+LocalGlyphRasterizer::LocalGlyphRasterizer(const optional<std::string> fontFamily)
+    : impl(std::make_unique<Impl>(fontFamily)) {
+}
+
+LocalGlyphRasterizer::~LocalGlyphRasterizer() {
+}
+
+bool LocalGlyphRasterizer::canRasterizeGlyph(const FontStack&, GlyphID glyphID) {
+    return util::i18n::allowsFixedWidthGlyphGeneration(glyphID) && impl->isConfigured();
+}
+
+Glyph LocalGlyphRasterizer::rasterizeGlyph(const FontStack&, GlyphID glyphID) {
+    Glyph glyph;
+    glyph.id = glyphID;
+
+    QFontMetrics metrics(impl->font);
+    Size size(metrics.width(glyphID), metrics.height());
+
+    glyph.metrics.width = size.width;
+    glyph.metrics.height = size.height;
+    glyph.metrics.left = 3;
+    glyph.metrics.top = -8;
+    glyph.metrics.advance = metrics.width(glyphID);
+
+    QImage image(QSize(size.width, size.height), QImage::Format_ARGB32_Premultiplied);
```

Comment:
Not a big deal, but could you use `QImage::Format_Alpha8` here and just render to alpha-only from the start, instead of rendering to ARGB32 and then pulling out the alpha?